internal/trivy/trivy.go

```diff
@@ -40,45 +41,44 @@ type Config struct {
 	Token string
 }
 
-type securityResponse struct {
+type parsedTrivyReport struct {
 	Results []struct {
 		Vulnerabilities []struct {
 			Severity string `json:"Severity"`
 		} `json:"Vulnerabilities"`
 	} `json:"Results"`
 }
 
-func (tc *Config) ScanManifest(manifestRefString, keppelToken string) (trivyReport securityResponse, returnedError error) {
-	//we don't allow Trivy to take more than 10 minutes on a single image (which is already an
-	//insanely generous timeout)
-	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Minute)
-	defer cancel()
-
+// TODO: use specific types intead of string arguments
+func (tc *Config) ScanManifest(ctx context.Context, keppelToken string, manifestRef models.ImageReference, format string) (report []byte, parsedReport parsedTrivyReport, returnedError error) {
 	requestURL := tc.URL
 	requestURL.Path = "/trivy"
-	requestURL.RawQuery = url.Values{"image": {manifestRefString}}.Encode()
+	requestURL.RawQuery = url.Values{
+		"image":  {manifestRef.String()},
+		"format": {format},
+	}.Encode()
 
 	req, err := http.NewRequestWithContext(ctx, http.MethodGet, requestURL.String(), http.NoBody)
 	if err != nil {
-		return securityResponse{}, err
+		return nil, parsedTrivyReport{}, err
 	}
 
 	req.Header.Set(TokenHeader, tc.Token)
 	req.Header.Set(KeppelTokenHeader, keppelToken)
 
 	resp, err := http.DefaultClient.Do(req)
 	if err != nil {
-		return securityResponse{}, err
+		return nil, parsedTrivyReport{}, err
 	}
 	defer resp.Body.Close()
 	respBody, err := io.ReadAll(resp.Body)
 	if err != nil {
-		return securityResponse{}, err
+		return nil, parsedTrivyReport{}, err
 	}
 	if resp.StatusCode != http.StatusOK {
-		return securityResponse{}, fmt.Errorf("trivy proxy did not return 200: %d %s", resp.StatusCode, respBody)
+		return nil, parsedTrivyReport{}, fmt.Errorf("trivy proxy did not return 200: %d %s", resp.StatusCode, respBody)
 	}
 
-	err = json.Unmarshal(respBody, &trivyReport)
-	return trivyReport, err
+	err = json.Unmarshal(respBody, &parsedReport)
+	return respBody, parsedReport, err
```

Comment:
Since we only need either the parsed or the unparsed manifest, depending on the callsite, I recommend having this function (`ScanManifest`) return only the unparsed form, and having a second function (`ScanManifestAndParse`) that returns the parsed form.